site/tests/app/entities/course/CourseMaterialTester.php

```diff
@@ -64,4 +72,40 @@ public function testLinkNoSections() {
         $this->assertEquals($details['hidden_from_students'], $course_material->isHiddenFromStudents());
         $this->assertEquals($details['priority'], $course_material->getPriority());
     }
+
+    public function testUpdateHistory() {
+        $details = [
+            'type' => CourseMaterial::LINK,
+            'path' => '/tmp/file.txt',
+            'release_date' => new \DateTime('9998-01-01 05:00:00'),
+            'hidden_from_students' => true,
+            'priority' => 2.4,
+            'uploaded_by' => "Dummy Uploader",
+            'uploaded_date' => new \DateTime('9998-01-01 05:00:00'),
+            'last_edit_by' => "Dummy User",
+            'last_edit_date' => new \DateTime('9998-01-01 05:00:00'),
```

Comment:
it looks like edit_date is not updated when it is first uploaded. ie. edit_date is null when we first upload file. I discussed with Barb about it, and it is fine. 
But uploaded_date and last_edit_date cannot be same, with the current implementation. So can you just fix the last_edit_date to some time after uploaded_date?